packages/client/mutations/AddPokerTemplateDimensionMutation.ts

```diff
@@ -61,15 +61,18 @@ const AddPokerTemplateDimensionMutation: StandardMutation<
       const {templateId} = variables
       const nowISO = new Date().toJSON()
       const defaultScale = store.get(SprintPokerDefaults.DEFAULT_SCALE_ID)
+      const template = store.get(templateId)
+      if (!defaultScale || !template) return
       const proxyTemplateDimension = createProxyRecord(store, 'PokerTemplateDimension', {
-        scaleId: SprintPokerDefaults.DEFAULT_SCALE_ID,
+        createdAt: nowISO,
+        name: `*New Dimension #${dimensionCount + 1}`,
         description: '',
+        scaleId: SprintPokerDefaults.DEFAULT_SCALE_ID,
         sortOrder,
-        name: `*New Dimension #${dimensionCount + 1}`,
-        createdAt: nowISO,
         templateId,
-        scale: defaultScale
```

Comment:
@tianrunhe this was the source of the "update inside an update" bug.
1. the field `scale` doesn't exist on a dimension, so that was making it short circuit
2. createProxyRecord only accepts an object where the values are not linked records (ie primitives), for linked records, you have to call `setLinkedRecord` like below
3. make sure to include all linked records you can! for example, if a component tries to grab the `dimension.template.name` and `template` isn't provided in the optimistic updater, it's gonna crash

Reply:
Oh I see! Thanks for sharing!